ircstates/server.py

```diff
@@ -747,6 +761,30 @@ def _handle_whoisuser(self, line: Line) -> Emit:
             user.realname = realname
         return emit
 
+    @line_handler(RPL_WHOISOPERATOR)
+    def _handle_whoisoper(self, line: Line):
+        emit = self._emit()
+        nickname = line.params[1]
+        lowered = self.casefold(nickname)
+        if lowered is None:
+            lowered = nickname
+
+        if lowered == self.nickname_lower:
+            emit.self = True
+            self.is_oper = True
+
+        if lowered in self.users:
+            self.users[lowered].is_oper = True
+            emit.user = self.users[lowered]
+
+    @line_handler(RPL_YOUREOPER)
```

Comment:
tracking this as state, rather than just having `RPL_YOUREOPER` in `numerics.py` for other stuff to use, means we'll want to also listen to us deopering. `MODE nick -o`? not sure

Reply:
Oh yes thats something I forgot